### ISSUE TITLE
Expose supported file types

### ISSUE THREAD

**Issue by Dids** [jfversluis/FilePicker-Plugin-for-Xamarin-and-Windows#24]
Most use cases will likely only want to support a very limited set of file types, so it would make sense to allow manually overriding the list of supported types.

For example, the supported types (UTIs) for iOS are defined here:
https://github.com/jfversluis/FilePicker-Plugin-for-Xamarin-and-Windows/blob/develop/src/Plugin.FilePicker/Plugin.FilePicker.iOS/FilePickerImplementation.cs#L116

In my case, I'd want to replace all of those with a single UTI string.

**Comment by jfversluis** [comment]
I've given this some thought. But this is very iOS specific right? It would be great to figure out some kind of cross-platform approach. A custom enum with certain filetype groups? Images, Sound, Documents, Everything. But then we would get in trouble whenever someone would want to allow just one extension..

**Comment by Dids** [comment]
I've forked the project and I'm working on a potential solution.

There are definitely difference between all the platforms, but the best I could come up with was to simply include the supported file types in platform-specific formats, since unsupported formats will be dropped by the platform-specific pickers anyway, it made more sense to me.

For example, this is how I'm currently only getting Markdown (.md) files:

```csharp
var pickedFile = await CrossFilePicker.Current.PickFile(new string[] { "net.daringfireball.markdown", "md", ".md" });
```

If you don't specify any file types, it'll use the existing defaults that were defined per-platform.

In this case, the first entry is an iOS (and macOS) compatible UTI, with the second one being a macOS compatible file extension (macOS supports both UTIs and file extensions) and lastly the last one being a UWP compatible file extension.

I'm still missing an Android implementation, as I haven't yet researched what formats it supports, but I'm assuming it'll be a similar process.

Let me know what you think, as I'm just doing this so that I can utilize this specific functionality ASAP. :)

**Comment by jfversluis** [comment]
Looks great! I think that is the most flexible way to do it.

So, this way you just specify the values for each platform that you want to support right? Meaning that all the filters for all the platforms will be applied on each separate platform?
In the above case, if I'm on iOS, still the filters for `"net.daringfireball.markdown", "md", ".md"`  will be applied, but since only one is valid, only that one has effect.

Is there a scenario where this will conflict? Making a applied filter for one platform behave unexpectedly on another platform? Or maybe throw exceptions when a filter is in a invalid format?

**Comment by Dids** [comment]
It looks like Android is definitely one of the platforms where this doesn't work as is, and requires a bit of special treatment.

Since Android expects the types to be mime-types instead, I'm simply checking for the `/` character, then separating each type into a regex-like string, which seems to be supported (eg. `image/*|text/*`).

I'm not able to test this on UWP, but a similar approach could work, where only types starting with a period would be included, as UWP expects type names to be formatted like this: `.txt`.

These might sound like hacks, because technically they are, but I'm fairly confident there won't be any edge cases this way.

**Comment by Dids** [comment]
Submitted the initial PR:
https://github.com/jfversluis/FilePicker-Plugin-for-Xamarin-and-Windows/pull/26

As mentioned, UWP is entirely untested, but I've briefly tested that iOS, macOS and Android seem to work.

**Comment by jfversluis** [comment]
Great, I'll see if I can test UWP some time soon and maybe create a sample app in the process. Thanks for the effort!

**Comment by dgboss** [comment]
@Dids - Thanks for your work on this! 
@jfversluis - I see the PR has been merged, any idea when you might be pushing out another Nuget package?

**Comment by jfversluis** [comment]
I was holding down because of #65 there seems to have snuck a bug in there. Didn't get around to it yet, unfortunately. If you really can't wait you can try the development feed: https://ci.appveyor.com/nuget/filepicker-plugin-for-xamarin

**Comment by kirillkostrov** [comment]
@jfversluis any news on this?

**Comment by vividos** [comment]
The PR from Dids was merged in NuGet package 2.0.106-beta, and I also added sample code, see MainPage.xaml.cs in the Forms sample project. When you're in an iOS project, you can access the UTType values directly, otherwise you have to get the strings to the Forms project somehow.
